src/main/kotlin/com/autonomousapps/internal/utils/gradleStrings.kt

```diff
@@ -141,50 +141,46 @@ internal fun ComponentIdentifier.resolvedVersion(): String? = when (this) {
  * Given [Configuration.getDependencies], return this dependency set as a set of identifiers, per
  * [ComponentIdentifier.toIdentifier].
  */
-internal fun DependencySet.toIdentifiers(
-  metadataSink: MutableMap<String, Boolean> = mutableMapOf()
-): Set<String> = mapNotNullToSet {
-  it.toIdentifier(metadataSink)
+internal fun DependencySet.toIdentifiers(): Set<Pair<String, Boolean>> = mapNotNullToSet {
+  it.toIdentifier()
 }
 
 internal fun Dependency.toCoordinates(): Coordinates? {
   val identifier = toIdentifier() ?: return null
   return when (this) {
-    is ProjectDependency -> ProjectCoordinates(identifier)
+    is ProjectDependency -> ProjectCoordinates(identifier.first)
     is ModuleDependency -> {
       resolvedVersion()?.let { resolvedVersion ->
-        ModuleCoordinates(identifier, resolvedVersion)
-      } ?: FlatCoordinates(identifier)
+        ModuleCoordinates(identifier.first, resolvedVersion)
+      } ?: FlatCoordinates(identifier.first)
     }
-    else -> FlatCoordinates(identifier)
+    else -> FlatCoordinates(identifier.first)
   }
 }
 
 /**
  * Given a [Dependency] retrieved from a [Configuration], return it as an identifier, per
  * [ComponentIdentifier.toIdentifier].
  */
-internal fun Dependency.toIdentifier(
-  metadataSink: MutableMap<String, Boolean> = mutableMapOf()
-): String? = when (this) {
+internal fun Dependency.toIdentifier(): Pair<String, Boolean>? = when (this) {
   is ProjectDependency -> {
     val identifier = dependencyProject.path
-    if (isJavaPlatform()) metadataSink[identifier] = true
-    identifier
+    val notMainFeature = isDependencyToJavaPlatform() || isDependencyToNonMainFeature()
+    Pair(identifier.intern(), notMainFeature)
   }
   is ModuleDependency -> {
     // flat JAR/AAR files have no group.
     val identifier = if (group != null) "${group}:${name}" else name
-    if (isJavaPlatform()) metadataSink[identifier] = true
-    identifier
+    val notMainFeature = isDependencyToJavaPlatform() || isDependencyToNonMainFeature()
+    Pair(identifier.intern(), notMainFeature)
   }
   is FileCollectionDependency -> {
     // Note that this only gets the first file in the collection, ignoring the rest.
     when (files) {
       is ConfigurableFileCollection -> (files as? ConfigurableFileCollection)?.from?.let { from ->
         from.firstOrNull()?.toString()?.substringAfterLast("/")
-      }
-      is ConfigurableFileTree -> files.firstOrNull()?.name
+      }?.let { Pair(it, false) }
+      is ConfigurableFileTree -> files.firstOrNull()?.name?.let { Pair(it, false) }
```

Comment:
I think we lost the `intern()` calls on these two types.

Reply:
Good catch. Sorry for that. Added them.